src/webshare.js

```diff
@@ -316,39 +505,147 @@ const webshare = {
+          chunks.push(topStreamsToEnrich.slice(i, i + chunkSize));
+        }
+
+        for (const chunk of chunks) {
+          console.log(`⏱️ Enriching chunk of ${chunk.length} TOP streams (chunk ${chunks.indexOf(chunk) + 1}/${chunks.length})`);
+          const chunkStart = Date.now();
+          
+          const chunkDetails = await Promise.all(
+            chunk.map(async (stream) => {
+              try {
+                const details = module.exports.getById ? await module.exports.getById(stream.ident, token) : null;
+                return { stream, details };
+              } catch (err) {
+                console.log(`getById failed for TOP stream ${stream.ident}:`, err.message);
+                return { stream, details: null };
+              }
+            }),
+          );
+          
+          // Apply enrichment to each stream in the chunk
+          chunkDetails.forEach(({ stream, details }) => {
+            if (details) {
+              // attach bitrate and dimensions for display/sorting
+              stream.bitrate = details.bitrate || null;
+              stream.width = details.width || null;
+              stream.height = details.height || null;
+              
+              // Recalculate resolution using three-level fallback with API data
+              stream.resolution = determineResolutionFallback(details.filename || stream.description, stream.fileSize, details.width, details.height);
+              
+              // Update name field to show correct resolution after enrichment
+              stream.name = `Webshare${stream.strongMatch ? " ✅" : ""} ${stream.resolution || ""}`;
+              
+              // Update speed hint with bitrate if available
+              if (stream.bitrate) {
+                const speedHint = formatRequiredSpeed(stream.bitrate);
+                if (speedHint) {
+                  // Replace existing speed estimate with bitrate-based one
+                  stream.description = stream.description.replace(/\n⚡ \d+(\.\d+)? Mbps/, `\n⚡ ${speedHint}`);
+                }
+              }
+            }
+          });
+          
+          console.log(`⏱️ TOP streams chunk completed in ${Date.now() - chunkStart}ms`);
+          
+          // Reduce delay between chunks for better performance
+          if (chunks.indexOf(chunk) < chunks.length - 1) {
+            await new Promise(resolve => setTimeout(resolve, 25)); // Reduced from 50ms to 25ms
+          }
+        }
+        
+        const enrichmentEnd = Date.now();
+        console.log(`⏱️ PERFORMANCE: TOP streams enrichment completed in ${enrichmentEnd - enrichmentStart}ms total`);
+      }
+    } catch (e) {
+      console.log("TOP streams enrichment error:", e.message);
+      // ignore enrichment errors
+    }
+        
+    const searchEnd = Date.now();
+    console.log(`⏱️ PERFORMANCE: Search completed in ${searchEnd - searchStart}ms total, returning ${filteredItems.length} streams`);
```

Comment:
For tracking how long something takes it would be better to use `console.time` and `console.timeEnd`. It requires less clutter in the code. But I'd like to think about the added logs a bit more, it seems you added a lot of them and I am not sure we need all of these outside of dev.